app/models/recipe_ingredient.rb

```diff
@@ -1,3 +1,10 @@
 class RecipeIngredient < ApplicationRecord
+  belongs_to :recipe
+  belongs_to :ingredient
 
+  def self.total_cost(array)
+    array.sum do |ingredient|
```

Comment:
Using `.sum` on an array like this is using Ruby to do the job of the database via ActiveRecord. 

Instead, we want to use the `.sum(:field)` method in AR. The `:field` in my example is referencing a specific field on a table. 
Specifically, for user story 3, if you're looknig for the total cost of ingredients on a recipe, I would implement this in the `Recipe` model, and use its `ingredients` association with this method, something like: 
```ruby
# Recipe model
def total_cost
   ingredients.sum(:cost)
end
```

This way we're using the association from this Recipe to its ingredients, and letting the database to the sum calculation for us.